File: website/blueprints/bank/routes/loans.py
from flask import Blueprint, request, jsonify
from models import Loan, db

loan_bp = Blueprint("loan", __name__, url_prefix="/loan")

# Get loans by playerID
@loan_bp.route("/<player_id>", methods=["GET"])
def get_loan(player_id):
    try:
        loan = Loan.query.filter_by(player_id=player_id).first()
        if not loan:
            return jsonify({"message": "Loan not found"}), 404
        return jsonify(loan.to_dict()), 200
    except Exception as e:
        return jsonify({"error": str(e)}), 500

# Update or create loan for a player
@loan_bp.route("/", methods=["POST"])
def update_or_create_loan():
    data = request.json
    player_id = data.get("playerID")
    loan_amount = data.get("loanAmount")

    try:
        loan = Loan.query.filter_by(player_id=player_id).first()

        if loan:
            loan.loan_amount = loan_amount
            loan.last_updated = db.func.now()
            db.session.commit()
        else:
            loan = Loan(player_id=player_id, loan_amount=loan_amount)
            db.session.add(loan)
            db.session.commit()

        return jsonify(loan.to_dict()), 200
    except Exception as e:
        return jsonify({"error": str(e)}), 500

File: website/blueprints/bank/routes/loans.js
const express = require("express");
const router = express.Router();
const Loan = require("../models/Loan");

// Get loans by playerID
router.get("/:playerID", async (req, res) => {
  try {
    const loan = await Loan.findOne({ playerID: req.params.playerID });
    if (!loan) return res.status(404).json({ message: "Loan not found" });
    res.json(loan);
  } catch (err) {
    res.status(500).json({ error: err.message });
  }
});

// Update or create loan for a player
router.post("/", async (req, res) => {
  const { playerID, loanAmount } = req.body;
  try {
    let loan = await Loan.findOne({ playerID });
    if (loan) {
      loan.loanAmount = loanAmount;
      loan.lastUpdated = Date.now();
      await loan.save();
    } else {
      loan = new Loan({ playerID, loanAmount });
      await loan.save();
    }
    res.json(loan);
  } catch (err) {
    res.status(500).json({ error: err.message });
  }
});

module.exports = router;
